Question: How much are nearby Hospitals and Ambulatory Surgical Centers (surgery centers) paid for the same procedure / in the same area?

In [1]:
import pandas as pd 
import numpy as np  
import matplotlib.pyplot as plt  


import seaborn as sns

In [2]:
%matplotlib inline

d1 = Medicare Provider Utilization and Payment Data: Physician and Other Supplier PUF CY2017

Note: in the hopes of avoiding memory problems, I am only reading in the following columns:
National Provider Identifier                                  
Last Name/Organization Name of the Provider                  
Entity Type of the Provider
City of the Provider                                         
Zip Code of the Provider                                     
State Code of the Provider                                   
Provider Type                                                
Place of Service                                             
HCPCS Code                                                   
HCPCS Description                                            
Number of Distinct Medicare Beneficiary/Per Day Services      
Average Medicare Allowed Amount                             
Average Medicare Payment Amount                             

In [22]:
pd.read_csv('c:/users/erinm/Documents/healthcare-bluebook-green-team/data/Medicare_Provider_Utilization_and_Payment_Data__Physician_and_Other_Supplier_PUF_CY2017.tsv', sep='\t').dtypes

National Provider Identifier                                  int64
Last Name/Organization Name of the Provider                  object
First Name of the Provider                                   object
Middle Initial of the Provider                               object
Credentials of the Provider                                  object
Gender of the Provider                                       object
Entity Type of the Provider                                  object
Street Address 1 of the Provider                             object
Street Address 2 of the Provider                             object
City of the Provider                                         object
Zip Code of the Provider                                     object
State Code of the Provider                                   object
Country Code of the Provider                                 object
Provider Type                                                object
Medicare Participation Indicator                

In [23]:
d1=pd.read_csv(
        'c:/users/erinm/Documents/healthcare-bluebook-green-team/data/Medicare_Provider_Utilization_and_Payment_Data__Physician_and_Other_Supplier_PUF_CY2017.tsv', 
        sep='\t',
        usecols=['National Provider Identifier', 
                 'Last Name/Organization Name of the Provider',
                 'Entity Type of the Provider',
                 'City of the Provider', 
                 'Zip Code of the Provider', 
                 'State Code of the Provider',
                 'Provider Type',
                 'Place of Service',
                 'HCPCS Code', 
                 'HCPCS Description', 
                 'Number of Distinct Medicare Beneficiary/Per Day Services', 
                 'Average Medicare Allowed Amount', 
                 'Average Medicare Payment Amount']
)     
d1.shape

(9847443, 13)

In [24]:
d1.columns = ['npi', 'last_name', 'entity_type', 'city', 'orig_zip', 'state', 'provider_type', 'place_of_service', 'hcpcs_code', 'hcpcs_desc', 'no_dist_ben_per_day_svcs', 'avg_med_allowed_amt', 'avg_med_pay_amt']

In [30]:
d1.loc[:,'zip']= d1['orig_zip'].astype(str).str[:5]
del d1['orig_zip']
d1.shape

(9847443, 13)

In [31]:
d1.head()

npi  last_name entity_type        city state      provider_type  \
0  1003000126  ENKESHAFI           I  CUMBERLAND    MD  Internal Medicine   
1  1003000126  ENKESHAFI           I  CUMBERLAND    MD  Internal Medicine   
2  1003000126  ENKESHAFI           I  CUMBERLAND    MD  Internal Medicine   
3  1003000126  ENKESHAFI           I  CUMBERLAND    MD  Internal Medicine   
4  1003000126  ENKESHAFI           I  CUMBERLAND    MD  Internal Medicine   

  place_of_service hcpcs_code  \
0                F      99217   
1                F      99218   
2                F      99219   
3                F      99220   
4                F      99221   

                                          hcpcs_desc  \
0                Hospital observation care discharge   
1     Hospital observation care typically 30 minutes   
2     Hospital observation care typically 50 minutes   
3  Hospital observation care typically 70 minutes...   
4  Initial hospital inpatient care, typically 30 ...   

   no_dist_ben_per_day_svcs  avg_med_allowed_amt  avg_med_pay_amt    zip  
0                       100            73.398800        56.827200  21502  
1                        26           100.080000        78.460000  21502  
2                        52           136.380000       102.807692  21502  
3                        59           190.363729       141.293559  21502  
4                        16           101.680000        79.710000  21502

In [32]:
d4 = pd.read_excel('c:/users/erinm/documents/healthcare-bluebook-green-team/data/ZIP_CBSA_032020.xlsx', dtype={'ZIP':'str'}, usecols=[0,1])
d4.head()

ZIP   CBSA
0  00501  35620
1  00601  38660
2  00602  10380
3  00603  10380
4  00604  10380

In [33]:
d4.columns = ['zip', 'cbsa']

In [34]:
provider_cbsas = pd.merge(d1, d4, on=('zip'), how='inner')
provider_cbsas

npi  last_name entity_type        city state  \
0         1003000126  ENKESHAFI           I  CUMBERLAND    MD   
1         1003000126  ENKESHAFI           I  CUMBERLAND    MD   
2         1003000126  ENKESHAFI           I  CUMBERLAND    MD   
3         1003000126  ENKESHAFI           I  CUMBERLAND    MD   
4         1003000126  ENKESHAFI           I  CUMBERLAND    MD   
...              ...        ...         ...         ...   ...   
10491617  1992990154      YOUNG           I     GILBERT    AZ   
10491618  1992990154      YOUNG           I     GILBERT    AZ   
10491619  1992990154      YOUNG           I     GILBERT    AZ   
10491620  1992990154      YOUNG           I     GILBERT    AZ   
10491621  1992990154      YOUNG           I     GILBERT    AZ   

                                   provider_type place_of_service hcpcs_code  \
0                              Internal Medicine                F      99217   
1                              Internal Medicine                F      99218   
2                              Internal Medicine                F      99219   
3                              Internal Medicine                F      99220   
4                              Internal Medicine                F      99221   
...                                          ...              ...        ...   
10491617  Physical Therapist in Private Practice                O      97110   
10491618  Physical Therapist in Private Practice                O      97112   
10491619  Physical Therapist in Private Practice                O      97140   
10491620  Physical Therapist in Private Practice                O      97162   
10491621  Physical Therapist in Private Practice                O      G0283   

                                                 hcpcs_desc  \
0                       Hospital observation care discharge   
1            Hospital observation care typically 30 minutes   
2            Hospital observation care typically 50 minutes   
3         Hospital observation care typically 70 minutes...   
4         Initial hospital inpatient care, typically 30 ...   
...                                                     ...   
10491617  Therapeutic exercise to develop strength, endu...   
10491618  Therapeutic procedure to re-educate brain-to-n...   
10491619  Manual (physical) therapy techniques to 1 or m...   
10491620  Evaluation of physical therapy, typically 30 m...   
10491621  Electrical stimulation (unattended), to one or...   

          no_dist_ben_per_day_svcs  avg_med_allowed_amt  avg_med_pay_amt  \
0                              100            73.398800        56.827200   
1                               26           100.080000        78.460000   
2                               52           136.380000       102.807692   
3                               59           190.363729       141.293559   
4                               16           101.680000        79.710000   
...                            ...                  ...              ...   
10491617                       338            27.452543        21.212526   
10491618                       116            31.496838        23.968309   
10491619                       329            22.761894        17.583320   
10491620                        18            81.520000        60.368889   
10491621                        99            10.310000         7.835152   

            zip   cbsa  
0         21502  19060  
1         21502  19060  
2         21502  19060  
3         21502  19060  
4         21502  19060  
...         ...    ...  
10491617  85246  38060  
10491618  85246  38060  
10491619  85246  38060  
10491620  85246  38060  
10491621  85246  38060  

[10491622 rows x 14 columns]

## Option 1 Stretch Goal (2nd part)
Filters for Physician/Other Supplier
Entity Type = I
Place of Service = O
HCPCS = 99213 (or any office visit – 99211-99215 / 99201-99205)

EA note: HCPCS 99211-15 are office/outpatient visits for ESTABLISHED patients, for 5, 10, 15, 25, & 40 min; 
         HCPCS 99201-05 are office/outpatient visits for NEW patients, for 10, 20, 30, 45, and 65 mins.

In [53]:
#constant filters
filter1 = provider_cbsas['entity_type'].isin(['I'])
filter2 = provider_cbsas['place_of_service'].isin(['O'])

constant_filters = provider_cbsas[filter1 & filter2]
constant_filters.shape

(6179882, 14)

In [36]:
#variable filter 3 (hcpcs)

#new patient HCPCS
#filter3_new = d1['hcpcs_code'].isin(['99201', '99202', '99203', '99204', '99205'])

#est patient HCPCS
filter3_est = constant_filters['hcpcs_code'].isin(['99211','99212','99213','99214','99215'])


hcpcs_filter = constant_filters[filter3_est]
hcpcs_filter.shape

(1083167, 14)

In [38]:
hcpcs_filter.head()

npi last_name entity_type        city state      provider_type  \
39  1003826983    FOSTER           I  CUMBERLAND    MD    Pain Management   
40  1003826983    FOSTER           I  CUMBERLAND    MD    Pain Management   
41  1003826983    FOSTER           I  CUMBERLAND    MD    Pain Management   
42  1003826983    FOSTER           I  CUMBERLAND    MD    Pain Management   
49  1003951625  MATYASIK           I  CUMBERLAND    MD  Pulmonary Disease   

   place_of_service hcpcs_code  \
39                O      99212   
40                O      99213   
41                O      99214   
42                O      99215   
49                O      99213   

                                           hcpcs_desc  \
39  Established patient office or other outpatient...   
40  Established patient office or other outpatient...   
41  Established patient office or other outpatient...   
42  Established patient office or other outpatient...   
49  Established patient office or other outpatient...   

    no_dist_ben_per_day_svcs  avg_med_allowed_amt  avg_med_pay_amt    zip  \
39                        13            45.280000        32.769231  21502   
40                        48            75.700000        57.361667  21502   
41                      3078           111.280000        81.841462  21502   
42                        54           149.770000       104.812593  21502   
49                       596            75.689279        53.417550  21502   

     cbsa  
39  19060  
40  19060  
41  19060  
42  19060  
49  19060

In [40]:
hcpcs_filter['hcpcs_desc'].value_counts()

Established patient office or other outpatient visit, typically 15 minutes    427247
Established patient office or other outpatient, visit typically 25 minutes    392403
Established patient office or other outpatient visit, typically 10 minutes    129055
Established patient office or other outpatient, visit typically 40 minutes    101891
Established patient office or other outpatient visit, typically 5 minutes      32571
Name: hcpcs_desc, dtype: int64

In [42]:
hcpcs_filter['provider_type'].value_counts()

Family Practice                                   176850
Internal Medicine                                 136543
Nurse Practitioner                                129187
Physician Assistant                                78970
Orthopedic Surgery                                 45547
                                                   ...  
Physical Therapist in Private Practice                 3
Chiropractic                                           3
Registered Dietitian or Nutrition Professional         2
Speech Language Pathologist                            1
Licensed Clinical Social Worker                        1
Name: provider_type, Length: 76, dtype: int64

Since 99203 (office/outpatient visit for 30 min) is the most frequent for new patients, I use that one first.
Compare to 99214 (estab patient, 25 min, which is 2nd most populous)

In [76]:
#variable filters (hcpcs, cbsas, provider type)

# HCPCS
filter_hcpcs_newpat = constant_filters['hcpcs_code'].isin(['99203'])
filter_hcpcs_estpat = constant_filters['hcpcs_code'].isin(['99214'])

#cbsas
filter_cbsa_nash = constant_filters['cbsa'].isin(['34980'])
filter_cbsa_atl = constant_filters['cbsa'].isin(['12060'])
filter_cbsa_chi = constant_filters['cbsa'].isin(['16980'])

#provider type
filter_prov_np = constant_filters['provider_type'].isin(['Nurse Practitioner'])
filter_prov_famprac = constant_filters['provider_type'].isin(['Family Practice'])

#new_nash_np = constant_filters[filter_hcpcs_newpat & filter_cbsa_nash & filter_prov_np]
#new_nash_famprac = [filter_hcpcs_newpat & filter_cbsa_nash & filter_prov_famprac]
new_atl_np = [filter_hcpcs_newpat & filter_cbsa_atl & filter_prov_np]
#new_atl_famprac =[filter_hcpcs_newpat & filter_cbsa_atl & filter_prov_famprac]
#new_chi_np = [filter_hcpcs_newpat & filter_cbsa_chi & filter_prov_np]
#new_chi_famprac = [[filter_hcpcs_newpat & filter_cbsa_chi & filter_prov_famprac]

#est_nash_np = constant_filters[filter_hcpcs_estpat & filter_cbsa_nash & filter_prov_np]
#est_nash_famprac = [filter_hcpcs_estpat & filter_cbsa_nash & filter_prov_famprac]
#est_atl_np = [filter_hcpcs_estpat & filter_cbsa_atl & filter_prov_np]
#est_atl_famprac = [filter_hcpcs_estpat & filter_cbsa_atl & filter_prov_famprac]
#est_chi_np = [filter_hcpcs_estpat & filter_cbsa_chi & filter_prov_np]
#est_chi_famprac = [filter_hcpcs_estpat & filter_cbsa_chi & filter_prov_famprac]

new_atl_np.head()


AttributeError: 'list' object has no attribute 'head'

In [103]:
#variable filters (hcpcs, cbsas, provider type)

# HCPCS
filter_hcpcs_estpat = constant_filters['hcpcs_code'].isin(['99214'])

#cbsas
filter_cbsa_nash = constant_filters['cbsa'].isin(['34980'])


#provider type
filter_prov_famprac = constant_filters['provider_type'].isin(['Nurse Practitioner'])

new_nash_famprac= constant_filters[filter_hcpcs_estpat & filter_cbsa_nash & filter_prov_famprac]

new_nash_famprac.shape

(679, 14)

In [104]:
new_nash_famprac['avg_med_allowed_amt'].describe()

count    679.000000
mean      86.099169
std        2.318339
min       64.380000
25%       86.308966
50%       86.620000
75%       86.620000
max       92.940000
Name: avg_med_allowed_amt, dtype: float64

In [85]:
new_nash_famprac['no_dist_ben_per_day_svcs'].describe()

count    323.000000
mean      33.235294
std       27.877511
min       11.000000
25%       15.000000
50%       23.000000
75%       39.000000
max      212.000000
Name: no_dist_ben_per_day_svcs, dtype: float64

The CBSA covering Nashville is 34980. 
CBSA 12060 (includes Atlanta)
CBSA 16980 (includes Chicago)


In [ ]:
nashville_cbsa = provider_crosswalk[provider_crosswalk['cbsa'].isin(['16980'])]
nashville_cbsa.shape

In [ ]:
nashville_cbsa.head(6)

In [ ]:
nashville_cbsa['provider_type'].value_counts()

nurse practitioner (nashville=304, atl = 300, chic = 460)  
family practice (nashville = , atl = 323, chic = 452 )